### PR TITLE
Fix for #33134

### PR DESCRIPTION
**Pull request by prabindh**
Updating softmax args description - logits can be of any type that can be passed to convert_to_tensor(), and associated errors. https://github.com/tensorflow/tensorflow/issues/33134

### CODE REVIEW

**Review by mihaimaruseac — APPROVED** [review]
Thank you

**Comment by mihaimaruseac** [comment]
On the other hand, this is only changing the documentation of a deprecated v1 function. As we are living in a 2.0 world, maybe this is not as useful?

Can you please make the changes to the v2 code if needed?
